### DISCUSSION TITLE
`t.headers.split` is not a function causing 500 status errors in prod

### DISCUSSION THREAD

**Discussion by adeniyii** [vercel/community#6953] [Bug Report] [answered]
### Page to Investigate

https://kepler-hazel.vercel.app/

### Steps to Reproduce

This bug shows up for already signed in users. 

```
Uncaught Exception: TypeError: t.headers.split is not a function
at _onRequestCreate (/var/task/apps/frontend/.next/server/chunks/6849.js:12:27026)
at Channel.publish (node:diagnostics_channel:143:9)
at new Request (node:internal/deps/undici/undici:6295:27)
at [dispatch] (node:internal/deps/undici/undici:9077:25)
at Intercept (node:internal/deps/undici/undici:8812:20)
at [Intercepted Dispatch] (node:internal/deps/undici/undici:5673:16)
at Client.dispatch (node:internal/deps/undici/undici:5689:44)
at [dispatch] (node:internal/deps/undici/undici:5920:32)
at [Intercepted Dispatch] (node:internal/deps/undici/undici:5666:33)
at Pool.dispatch (node:internal/deps/undici/undici:5689:44)
Node.js process exited with exit status: 129. The logs above can help with debugging the issue.
```

Non-signed in users get a `405 not allowed` when they try to sign-in.

### Repro
- Go to provided URL
- Sign up
- Try to sign-in

### More context
We use `Supabase` for managing auth, and we get a 403 error in their auth logs: `"error":"403: invalid claim: missing sub claim"`.

This issue showed up earlier today, but seemed to resolve itself.
It's back now, and we would appreciate some help in debugging this cryptic failure.

### Update
Downgrading our node version to 18 (from 20) seems to have fixed it.
To be clear though, we have been on node 20 for months without issue. 
So we still don't know why it suddenly turned problematic.

**Comment by thiagomorales** [comment, marked as answer]
Hi! I had the same errors with serverless functions in Vercel, but it was related to a dependency on Sentry.

Please check if your problem is related to this issues: 
- https://github.com/getsentry/sentry-javascript/issues/12218
- https://github.com/getsentry/sentry-javascript/issues/10936

The Vercel team updated their runtime yesterday exposing it to this issue.

Regards.

**Comment by SkycladObserver** [comment]
I experienced this as well. I fixed this issue by updating sentry to `7.106.0` as suggested by other threads.

**Reply by lutfiarahmanda** [reply]
i thought it was next-auth.js because it return 405, after updating the @sentry/nextjs package from 7.104.0 to 7.116.0 it fixed the problem
